android-pdf-viewer/src/main/java/com/github/barteksc/pdfviewer/annotation/core/annotations/Annotation.kt

```diff
@@ -8,7 +8,10 @@ import com.github.barteksc.pdfviewer.annotation.core.shapes.convertCoordinatesFr
 
 /**  Points are in the order: topLeft, topRight, bottomRight, bottomLeft */
 data class Annotation(
-    val type: String, val points: List<PointF>, val relations: Relations? = null
+    val type: String,
+    val points: List<PointF>,
+    val relations: Relations? = null,
+    val uri: String? = null
```

Comment:
Does it make sense to have it here? We are mixing types this way. What would happen if we add Text annotation for example, do we just add a new text property to the class?

Reply:
I guess when I do the cast, the uri is not set so I'm not sure how to avoid having `uri` in the `Annotation`. Use interface or abstract class?

Reply:
> Tried to make it better by creating LinkAnnotation

I would expect the base class to hold the shared properties only. Currently you have `relations` in the base class which is something thats special to measureLib annotation thus making us to use `relations=null` in the subclass (LinkAnnotation).

Also `points` is an overriden value which is not nessecary you can just have `LinkAnnotation` like

```kotlin
class LinkAnnotation(points: List<PointF>, val uri: String?) :
    Annotation(type = AnnotationType.LINK.name, points, relations = null)
```

Ultimately base class should hold just the `type` and the `points`. relations is special to measureLib annotation and uri is special to link annotation so these should be inside the subclasses.

>  guess when I do the cast, the uri is not set so I'm not sure how to avoid having uri in the Annotation.

Try setting it then before casting?

Reply:
> Ultimately base class should hold just the type and the points

Will do that
 
> Try setting it then before casting?

Not sure what you mean exactly

I think the problem is that `PdfUtil.getAnnotationsFrom` returns `Annotation` so the uri property is cut off but I'm not sure how to handle that

Reply:
> I think the problem is that PdfUtil.getAnnotationsFrom returns Annotation so the uri property is cut off but I'm not sure how to handle that

I don't think this is an issue take this for example

https://pl.kotl.in/uNAXy3dVa

As you can see even though the list is of type 'A' the subclass parameter is properly kept so something else is going on. 

Have you tried debugging the flow?

Reply:
Thanks for the example

Thought the problem was that `getExtractedLinkAnnotation` still has return type `Annotation` so changed that but it didn't help


Then I thought it was picking the wrong annotation

<img width="1317" alt="Screenshot 2024-03-27 at 11 33 00" src="https://github.com/VERO-Digital-Solutions/AndroidPdfViewerV2/assets/57250076/54a2b7d6-b7e1-4c12-a26b-c8189ea491aa">

So I made a change so we extract only the link annotations that contain URI and I was able to see the link opened on click 

Then decided to do `filter` instead of `firstOrNull` and I see more annotations being detected

![image](https://github.com/VERO-Digital-Solutions/AndroidPdfViewerV2/assets/57250076/15971450-c644-4f3e-98bb-78378ae65802)

<img width="252" alt="Screenshot 2024-03-27 at 12 33 45" src="https://github.com/VERO-Digital-Solutions/AndroidPdfViewerV2/assets/57250076/86bbf1e9-d682-47d1-8608-20fc99d55bc1">



Extracted `relations` out of the base `Annotation` too